src/Microsoft.OData.Core/JsonLight/ODataJsonLightParameterDeserializer.cs

```diff
@@ -216,5 +222,192 @@ internal bool ReadNextParameter(PropertyAndAnnotationCollector propertyAndAnnota
 
             return parameterRead;
         }
+
+        /// <summary>
+        /// Reads the next parameter from the parameters payload.
+        /// </summary>
+        /// <param name="propertyAndAnnotationCollector">The duplicate property names checker used to read a parameter payload.</param>
+        /// <returns>
+        /// A task that represents the asynchronous read operation.
+        /// The value of the TResult parameter contains true if a parameter was read from the payload; otherwise false.
+        /// </returns>
+        /// <remarks>
+        /// Pre-Condition:  Property or EndObject   the property node of the parameter to read or the end object node if there are not parameters
+        /// Post-Condition: Property or EndObject   the node after the property value of a primitive, complex or null collection parameter
+        ///                 Any                     the start of the value representing a non-null collection parameter (the collection reader will fail if this is not a StartArray node)
+        /// </remarks>
+        internal async Task<bool> ReadNextParameterAsync(PropertyAndAnnotationCollector propertyAndAnnotationCollector)
+        {
+            Debug.Assert(propertyAndAnnotationCollector != null, "propertyAndAnnotationCollector != null");
+            this.AssertJsonCondition(JsonNodeType.Property, JsonNodeType.EndObject);
+
+            bool parameterRead = false;
+            if (this.JsonReader.NodeType == JsonNodeType.Property)
+            {
+                bool foundCustomInstanceAnnotation = false;
+                await this.ProcessPropertyAsync(
+                    propertyAndAnnotationCollector,
+                    propertyAnnotationValueReaderAsync,
+                    async (propertyParsingResult, parameterName) =>

```

Comment:
A good candidate for a future update on this lambda so it captures less items. :smile:

Reply:
I plan to tackle in a separate to address the same concern in related PRs